ipatests/test_integration/test_automember.py

```diff
@@ -0,0 +1,316 @@
+        self.master.run_command(['ipa', 'user-del', 'user1'])
+        result = self.master.run_command(
+            ['ipa', 'group-show', 'HoustonUsers'],
+        )
+        assert "user1" not in result.stdout_text
+
+        '#testcase cleanup'
+        cmdlist = [['ipa', 'automember-del', '--type=group', 'DallasUsers'],
+                   ['ipa', 'automember-del', '--type=group', 'HoustonUsers'],
+                   ['ipa', 'group-del', 'DallasUsers'],
+                   ['ipa', 'group-del', 'HoustonUsers']]
+
+        for cmd in cmdlist:
+            self.master.run_command(cmd,)
+
+    def test_modify_host_entry_unmembership(self):
+        # https://pagure.io/389-ds-base/issue/50077
+
+        '#testcase setup'
+        cmdlist = [['ipa', 'hostgroup-add', '--desc="Brno"', 'Brno'],
+                   ['ipa', 'automember-add', '--type=hostgroup', 'Brno'],
+                   ['ipa', 'automember-add-condition',
+                    '--key=nshostlocation', '--type=hostgroup',
+                    '--inclusive-regex=Brno', 'Brno'],
+                   ['ipa', 'host-add', 'host1.example.com',
+                    '--location=Brno', '--force']]
+
+        for cmd in cmdlist:
+            self.master.run_command(cmd,)
+
+        result = self.master.run_command(
+            ['ipa', 'hostgroup-show', 'Brno'],
+        )
+        assert "host1" in result.stdout_text
+
+        '#Modifying the user group so that it becomes part of new group'
+        cmdlist = [['ipa', 'hostgroup-add', '--desc="Boston"', 'Boston'],
+                   ['ipa', 'automember-add', '--type=hostgroup', 'Boston'],
+                   ['ipa', 'automember-add-condition',
+                    '--key=nshostlocation', '--type=hostgroup',
+                    '--inclusive-regex=Boston', 'Boston'],
+                   ['ipa', 'host-mod', 'host1.example.com',
+                    '--location=Boston']]
+
+        for cmd in cmdlist:
+            self.master.run_command(cmd,)
+
+        result = self.master.run_command(
+            ['ipa', 'hostgroup-show', 'Boston'],
+        )
+        assert "host1" in result.stdout_text
+
+        result = self.master.run_command(
+            ['ipa', 'hostgroup-show', 'Brno'],
+        )
+        assert "host1" not in result.stdout_text
+
+        '#Deleting the user entry and checking its not part of group anymore'
+
+        self.master.run_command(['ipa', 'host-del', 'host1.example.com'])
+        result = self.master.run_command(
+            ['ipa', 'hostgroup-show', 'Boston'],
+        )
+        assert "host1" not in result.stdout_text
+
+        '#testcase cleanup'
+        cmdlist = [['ipa', 'automember-del', '--type=hostgroup', 'Boston'],
+                   ['ipa', 'automember-del', '--type=hostgroup', 'Brno'],
+                   ['ipa', 'hostgroup-del', 'Brno'],
+                   ['ipa', 'hostgroup-del', 'Boston']]
+
+        for cmd in cmdlist:
+            self.master.run_command(cmd,)
```

Comment:
For clean, I would add raiseonerr=False to make sure all the cmds are executed.
The test can also be written with
try:
[do calls]
finally:
[do cleanup]